library/src/main/java/com/novoda/downloadmanager/DownloadBatchStatusNotificationDispatcher.java

```diff
@@ -30,6 +32,15 @@ void updateNotification(DownloadBatchStatus downloadBatchStatus) {
     }
 
     void setDownloadService(DownloadService downloadService) {
-        notificationDispatcher.setDownloadService(downloadService);
+        if (downloadService instanceof Service) {
```

Comment:
Adding a new interface that `DownloadService` and `MigrationService` extends from may alleviate instance check issue.